### PR TITLE
Reverse colors instead of values in heatmap legend for generic assay profile

### PR DESCRIPTION
**Pull request by pvannierop**
This PR will:
- Setting `ASC` or `DESC` as `value_sort_order` in the meta file of an generic assay profile will reverse the colors, not the values in the legend.
- Cleanup and correct code for creating the legend.

Note: documentation is improved in PR https://github.com/cBioPortal/cbioportal/pull/8016.

# Old legend
image

# New legend
image

### CODE REVIEW

**Comment by alisman** [comment]
@dippindots @pvannierop should we merge this?  i'm confused now.

**Review by dippindots — APPROVED** [review]
Thanks! LGTM, just have one text suggestion.

**Comment by dippindots** [comment]
@alisman We should revert the previous pr I made, and merge @pvannierop's pr (both `frontend` and `backend`).

We should do these:
1. merge[ this pr](https://github.com/cBioPortal/cbioportal-frontend/pull/3481) which revert the previous changes
2. @pvannierop should rebase to the latest master and solve conflict if any
3. @pvannierop may need to update some related screenshots
4. merge this pr and [backend pr](https://github.com/cBioPortal/cbioportal/pull/8016)

File: src/shared/components/oncoprint/OncoprintUtils.ts [hunk 1]
```diff
@@ -234,33 +245,30 @@ export function getGenericAssayTrackRuleSetParams(
...
-    //  along side other track specs (if possible)
-    // - When the most extreme value does not reach the pivotThreshold the pivotThreshold is used a most extreme value
+    //      1. Smaller values are `important` and darker blue (a.k.a. ASC sort order)
+    //      2. Larger values are `important` and darker blue (a.k.a. DESC sort order)
+    // - The pivot threshold denotes an arbitrary boundary between important (in red)
+    //   and unimportant (in blue) values. When a pivot threshold is defined blue
+    //   and red gradient to white at the pivotThreshold value.
+    // - The most extreme value in the legend is should be the largest value in the
+    //   current track group. It is passed in alongside other track specs (if possible).
+    // - When the most extreme value does not reach the pivotThreshold the pivotThreshold,
```
[the remark this answers is not in the document]
Comment (pvannierop) [reply src/shared/components/oncoprint/OncoprintUtils.ts]: @dippindots Reworded the comment to make it more clear.

**Review by adamabeshouse — APPROVED** [review]

**Pull request merged**